tests/foreman/api/test_classparameters.py

```diff
@@ -15,78 +15,188 @@
+        },
+        {
+            u'sc_type': 'integer',
+            u'value': gen_string('utf8'),
+        },
+        {
+            u'sc_type': 'real',
+            u'value': gen_string('alpha'),
+        },
+        {
+            u'sc_type': 'array',
+            u'value': '0',
+        },
+        {
+            u'sc_type': 'hash',
+            u'value': 'a:test',
+        },
+        {
+            u'sc_type': 'yaml',
+            u'value': '{a:test}',
+        },
+        {
+            u'sc_type': 'json',
+            u'value': gen_string('alpha'),
+        },
+    ]
+
+
 class SmartClassParametersTestCase(APITestCase):
     """Implements Smart Class Parameter tests in API"""
 
+    @classmethod
+    def setUpClass(cls):
+        super(SmartClassParametersTestCase, cls).setUpClass()
+        cls.host_name = settings.server.hostname
+        cls.host = entities.Host().search(
+            query={'search': 'name=' + cls.host_name}
+        )[0]
+        ssh.command('puppet module install --force puppetlabs/ntp')
+        cls.env = entities.Environment().search(
+            query={'search': 'name="production"'}
+        )[0]
+        cls.proxy = entities.SmartProxy(name=cls.host_name).search()[0]
+        cls.proxy.import_puppetclasses(environment=cls.env)
+        cls.puppet = entities.PuppetClass().search(
+            query={'search': 'name="ntp"'}
+        )[0]
+        cls.sc_params_list = entities.SmartClassParameters().search(
+            query={'search': 'puppetclass="ntp"', 'per_page': 100}
+        )
+        if len(cls.sc_params_list) < 45:
+            raise RuntimeError('There are not enough smart class parameters to'
+                               ' work with in provided puppet class')
+
     @run_only_on('sat')
-    @stubbed()
-    @tier3
+    @tier2
     def test_positive_list_parameters_by_host_id(self):
         """List all the parameters included in specific Host by its id.
 
         @id: a9e551f9-261b-40e6-b7f6-35621fc46285
 
         @assert: Parameters listed for specific Host.
 
-        @caseautomation: notautomated
-
-        @CaseLevel: System
+        @CaseLevel: Integration
         """
+        sc_param = self.sc_params_list.pop()
+        sc_param.override = True
+        sc_param.update(['override'])
+        self.assertEqual(sc_param.read().override, True)
+        self.host.puppet_class = [self.puppet]
+        self.host.update(['puppet_class'])
+        self.assertGreater(len(self.host.list_scparams()), 0)
 
     @run_only_on('sat')
-    @stubbed()
-    @tier3
+    @tier2
     def test_positive_list_parameters_by_hostgroup_id(self):
         """List all the parameters included in specific HostGroup by id.
 
         @id: 88ceea89-b8b5-4ca2-9d59-3b2614c7f9a7
 
         @assert: Parameters listed for specific HostGroup.
 
-        @caseautomation: notautomated
-
-        @CaseLevel: System
+        @CaseLevel: Integration
         """
+        sc_param = self.sc_params_list.pop()
```

Comment:
I'm going to agree with @jyejare here. Since this list is finite, what happens when we add a new test and there's nothing left to 'pop'? If the exercise is to verify that all types of parameters work, that should be a test on its own. For everything else, a random value off that list should suffice.